### PR TITLE
Add support of namespace property to support Android Gradle Plugin 8.

### LINKED ISSUE #839: Add support of namespace property to support Android Gradle Plugin (AGP) 8
## Plugin Version

Latest (3.0.0).

## Steps to Reproduce

Upgrade your Android project to use AGP 8. In your `build.gradle` :

```gradle
buildscript {
  // ...

  dependencies {
    classpath 'com.android.tools.build:gradle:8.0.1'
    // ...
  }
}
```

Also don't forget to add a namespace in the `android` section of `app/build.gradle` :

```gradle
android {
  // ...
  namespace 'your.namespace'
}
```

Then `flutter run` to an Android device (or even `flutter build appbundle`).

**Expected results:** 

My app to compile fine.

**Actual results:** 

<details>
  <summary>Stacktrace</summary>

```
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':google_mobile_ads'.
> Could not create an instance of type com.android.build.api.variant.impl.LibraryVariantBuilderImpl.
   > Namespace not specified. Please specify a namespace in the module's build.gradle file like so:

     android {
         namespace 'com.example.namespace'
     }

     If the package attribute is specified in the source AndroidManifest.xml, it can be migrated automatically to the namespace value in the build.gradle file using the AGP Upgrade Assistant; please refer to https://developer.android.com/studio/build/agp-upgrade-assistant for more information.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s
Running Gradle task 'bundleRelease'...                           2 150ms
Gradle task bundleRelease failed with exit code 1
```

```
```

```
```

```
```

</details>

## More info

See https://github.com/flutter/flutter/issues/125181. It shouldn't be too hard to fix (see the PR I've sent here : https://github.com/bluefireteam/audioplayers/pull/1503/files), but since this project seems to use a custom `build.gradle`, I'm not sure if we can "just" copy and paste it.

### PR DESCRIPTION
**Pull request by Skyost**
## Description

This PR adds support of namespace property to support Android Gradle Plugin (AGP) 8. It fixes the following error :

```
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':google_mobile_ads'.
> Could not create an instance of type com.android.build.api.variant.impl.LibraryVariantBuilderImpl.
   > Namespace not specified. Please specify a namespace in the module's build.gradle file like so:

     android {
         namespace 'com.example.namespace'
     }

     If the package attribute is specified in the source AndroidManifest.xml, it can be migrated automatically to the namespace value in the build.gradle file using the AGP Upgrade Assistant; please refer to https://developer.android.com/studio/build/agp-upgrade-assistant for more information.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s
Running Gradle task 'bundleRelease'...                           2 150ms
Gradle task bundleRelease failed with exit code 1
```

## Related Issues

Fixes #839.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/googleads/googleads-mobile-flutter/issues
[Contributor Guide]: https://github.com/googleads/googleads-mobile-flutter/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Review by ps6067966 — APPROVED** [review]
LGTM

**Comment by gigafunk** [comment]
I am a newb struggling with this, i guess i just need to downgrade my gradle and keep an eye on this PR to try it when accepted?

**Comment by Skyost** [comment]
@gigafunk Yep.

**Review by LTPhantom — CHANGES_REQUESTED** [review]
It seems that this PR conflicts with a change in the Head. Can you rebase?

**Comment by Skyost** [comment]
@LTPhantom Done.

**Review by LTPhantom — APPROVED** [review]
Thanks for contributing to this!

File: packages/google_mobile_ads/example/android/app/build.gradle [hunk 1]
```diff
@@ -27,6 +27,8 @@ apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 android {
     compileSdkVersion 33
 
+    namespace 'io.flutter.plugins.googlemobileadsexample'
+
```
Comment (LTPhantom) [inline packages/google_mobile_ads/example/android/app/build.gradle:31]: Actually, can you update Java version from 11 to 17 in lines 54 and 55?
```
sourceCompatibility JavaVersion.VERSION_17
targetCompatibility JavaVersion.VERSION_17
```

**Comment by Skyost** [comment]
Done 👌

**Review by LTPhantom — APPROVED** [review]

File: packages/google_mobile_ads/android/build.gradle [hunk 2]
```diff
@@ -41,13 +45,12 @@ android {
       testImplementation 'org.mockito:mockito-inline:3.9.0'
       testImplementation 'org.robolectric:robolectric:4.4'
       testImplementation 'androidx.test:core:1.3.0'
```
Comment (LTPhantom) [inline packages/google_mobile_ads/android/build.gradle:47]: Please update the dependencies:
```
api 'com.google.android.gms:play-services-ads:22.4.0'
implementation 'com.google.android.ump:user-messaging-platform:2.1.0'
implementation 'androidx.constraintlayout:constraintlayout:2.1.4'
implementation 'androidx.lifecycle:lifecycle-process:2.6.2'
implementation 'com.google.errorprone:error_prone_annotations:2.16'
testImplementation 'junit:junit:4.13.2'
testImplementation 'org.hamcrest:hamcrest:2.2'
testImplementation 'org.mockito:mockito-inline:5.2.0'
testImplementation 'org.robolectric:robolectric:4.10.3'
testImplementation 'androidx.test:core:1.5.0'
```

File: packages/google_mobile_ads/example/android/app/build.gradle [hunk 3]
```diff
@@ -49,8 +51,8 @@ android {
     }
```
Comment (LTPhantom) [inline packages/google_mobile_ads/example/android/app/build.gradle:51]: add `multiDexEnabled true` in line 43

**Review by LTPhantom — CHANGES_REQUESTED** [review]
We are almost there, there are some tests that are failing and some extra changes are needed.

Besides the comment below, I would ask to change the `java-version` from 11 to 17 in the `.github/workflows/google_mobile_ads.yaml` file.

**Comment by Skyost** [comment]
Let's do that !

**Review by LTPhantom — APPROVED** [review]

**Comment by PedroContipelli** [comment]
When will this be in release? google_mobile_ads 3.2.0?

**Comment by nathantaal** [comment]
@PedroContipelli in the meanwhile you could use https://github.com/nathantaal/googleads-mobile-flutter, like this:

```

**Pull request merged**